fireant/slicer/dimensions.py

```diff
@@ -0,0 +1,256 @@
+        """
+        Creates a filter to filter a slicer query.
+
+        :param values:
+            An iterable of value to constrain the slicer query results by.
+
+        :return:
+            A slicer query filter used to filter a slicer query to results where this dimension is *not* one of a set of
+            values. Opposite of #isin.
+        """
+        return ExcludesFilter(self.definition, values)
+
+
+class UniqueDimension(Dimension):
+    """
+    This is a dimension that represents a field in a database which is a unique identifier, such as a primary/foreign
+    key. It provides support for a display value field which is selected and used in the results.
+    """
+
+    def __init__(self, key, label=None, definition=None, display_definition=None):
+        super(UniqueDimension, self).__init__(key=key, label=label, definition=definition)
+
+        self.display_definition = display_definition
+
+        self.display_key = '{}_display'.format(key) \
+            if display_definition is not None \
+            else None
+
+    def __hash__(self):
+        if self.has_display_field:
+            return hash('{}({},{})'.format(self.__class__.__name__, self.definition, self.display_definition))
+        return super(UniqueDimension, self).__hash__()
+
+    def isin(self, values, use_display=False):
+        """
+        Creates a filter to filter a slicer query.
+
+        :param values:
+            An iterable of value to constrain the slicer query results by.
+        :param use_display:
+            When True, the filter will be applied to the Dimesnion's display definition instead of the definition.
+
+        :return:
+            A slicer query filter used to filter a slicer query to results where this dimension is one of a set of
+            values. Opposite of #notin.
+        """
+        if use_display and self.display_definition is None:
+            raise QueryException('No value set for display_definition.')
+        filter_field = self.display_definition if use_display else self.definition
+        return ContainsFilter(filter_field, values)
+
+    def notin(self, values, use_display=False):
+        """
+        Creates a filter to filter a slicer query.
+
+        :param values:
+            An iterable of value to constrain the slicer query results by.
+        :param use_display:
+            When True, the filter will be applied to the Dimesnion's display definition instead of the definition.
+
+        :return:
+            A slicer query filter used to filter a slicer query to results where this dimension is *not* one of a set of
+            values. Opposite of #isin.
+        """
+        if use_display and self.display_definition is None:
+            raise QueryException('No value set for display_definition.')
+        filter_field = self.display_definition if use_display else self.definition
+        return ExcludesFilter(filter_field, values)
+
+    def wildcard(self, pattern):
```

Comment:
It would be good to rename this to `like` to fit with the other filter patterns which make the connection to the filter feel more natural.